### ISSUE TITLE
Recursor for parametric type seems broken

### ISSUE THREAD

**Issue by konn** [RedPRL/sml-redprl#694]
For a non-parametric data-type, we can use recursor instead of the `elim` tactic as follows:

```redprl
data BraceInt: (U 0 kan) {
  brace int
} by { auto }.

theorem UnbraceInt :  (-> (.BraceInt type ) int) by {
  lam b => `(.BraceInt rec [_] int b [z] z)
}.
```

On the other hand, it seems that recursors for parametric data-types don't work.
Consider the following parametric version of the above `Brace`

```redprl
data Brace [a: (U 0 coe)]: (U 0 kan) {
  brace a
} by { auto }.

theorem Unbrace: (-> [a: (U 0 coe)] (.Brace type a) a)
by {
  lam a b =>`(.Brace rec [_] a b [z] z);
  ?unresolved
}.
```

Then, RedPRL reports that there is an unresolved goal as follows:

```
list.prl:14.3-14.14 [Info]:
  ?unresolved.
  Goal #0.
    a : (U 0 coe)
    b : (. Brace type a)
    -----------------------------------
    (mem a (. Brace rec [_] a b [z] z))

list.prl:12.3-14.14 [Warning]:
  1 Remaining Obligations
```

Things won't change if one specialise type argument `a` in `Unbrace` theorem to `int`.

**Comment by favonia** [comment]
There is a bug in the arity calculation and will be fixed in #694. However, the `Unbrace` theorem is not obviously provable because the type `a` needs to handle the free `hcom`s added in the `Brace` type. In other words, it probably needs to be `kan`.

**Comment by konn** [comment]
I confirmed that the warnings above vanished after rebuilding `redprl` based on `inductive-elim` branch and replace `coe` with `kan`.
Thank you for your rapid reaction!
